core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -58,7 +59,10 @@ private[deploy] class Master(
   private val appIdPattern = conf.get(APP_ID_PATTERN)
 
   // For application IDs
-  private def createDateFormat = new SimpleDateFormat("yyyyMMddHHmmss", Locale.US)
+  private lazy val dateTimeFormatter =
```

Comment:
nit: Does not need to be `lazy` and can be moved to `object Master`, same in other places as well.

Reply:
+1